google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/project/DefaultAppEngineProjectService.java

```diff
@@ -366,48 +367,41 @@ public void generateDockerfile(AppEngineFlexibleDeploymentArtifactType type, Mod
   }
 
   @Nullable
-  private PsiElement generateFromTemplate(String templateName, String templateDirectoryName,
-      String fileName, Properties templateProperties, Module module) {
+  private PsiElement generateFromTemplate(String templateName, String outputFileName,
+      Path outputFolderPath, Properties templateProperties, Module module) {
     FileTemplate configTemplate = FileTemplateManager.getDefaultInstance().getInternalTemplate(
         templateName);
-    VirtualFile virtualFile = ModuleRootManager.getInstance(module).getContentRoots()[0]
-        .findFileByRelativePath("src/main");
-
-    if (virtualFile != null) {
-      PsiDirectory directory = PsiManager.getInstance(module.getProject())
-          .findDirectory(virtualFile);
-
-      if (directory != null) {
-        PsiDirectory configDirectory;
-        try {
-          directory.checkCreateSubdirectory(templateDirectoryName);
-          configDirectory = directory.createSubdirectory(templateDirectoryName);
-        } catch (IncorrectOperationException ioe) {
-          // checkCreateSubdirectory threw an exception suggesting that the directory may already
-          // exist. Skip creating the directory and attempt to write file.
-          configDirectory = directory.findSubdirectory(templateDirectoryName);
-        }
 
-        if (configDirectory != null
-            && FileTemplateUtil.canCreateFromTemplate(
-            new PsiDirectory[]{configDirectory}, configTemplate)) {
-          try {
-            return FileTemplateUtil.createFromTemplate(
-                configTemplate,
-                fileName,
-                templateProperties,
-                configDirectory);
-          } catch (Exception e) {
-            // If the file already exists, this exception will be thrown by createFromTemplate
-            // We want to silently skip the generation in this case.
-            logger.debug("Failed to create app.yaml from template. " + e.getMessage());
-          }
-        } else {
-          logger.error("Failed to create app.yaml from template in directory: " + configDirectory);
-        }
+    File outputFolder = outputFolderPath.toFile();
+    if (!outputFolder.exists() && !outputFolder.mkdirs()) {
+        logger.warn("Failed to create " + outputFolder.toString());
```

Comment:
minor, but maybe mention that its the directory that failed to create, in case outputFolder isn't useful